### PR TITLE
Feature/auto devserver port

### LINKED ISSUE #1545: Handle Vite port change
### Is your feature request related to a problem? Please describe.

Vite tries to use port 3000 by default and if it doesn't find it, will keep incrementing the port until it finds a free one. This means that the value in `frontend:dev:serverUrl` is mismatched and will fail in weird ways. 

### Describe the solution you'd like

A value for `frontend:dev:serverUrl` of `auto` would indicate that we should read up the vite port from the stdout of the vite process and use that.

### Describe alternatives you've considered

_No response_

### Additional context

Run docusaurus using `npm run start` and then try `wails dev` in a project. It results in the application becoming unresponsive.

### PR DESCRIPTION
**Pull request by leaanthony**
This PR adds an "auto" option for `frontend:dev:serverUrl` that attempts to determine the URL vite is running on via the vite output.
Updated default templates.
Fix package/variable name clashes.

Fixes #1545

### CODE REVIEW

File: v2/cmd/wails/internal/commands/dev/dev.go [hunk 1]
```diff
@@ -438,12 +450,32 @@ func runFrontendDevWatcherCommand(cwd string, devCommand string) (func(), error)
...
+				_, err := url.Parse(viteServerURL)
+				if err != nil {
+					LogRed(err.Error())
+					break
+				}
+				break
+			}
+		}
+	}
+
```
Comment (stffabi) [inline v2/cmd/wails/internal/commands/dev/dev.go:472]: According to the documentation of [stdout pipe](https://pkg.go.dev/os/exec#Cmd.StdoutPipe) one should call `wait` after all reads from the pipe have been completed.
So I would assume we shouldn't start the go routine which waits before the search has been finished.

Comment (stffabi) [inline v2/cmd/wails/internal/commands/dev/dev.go:472]: Is it fine to not fully consume the stdout pipe? IIRC on linux a command might block when the stdout's buffer is full, but I'm not sure if this is a problem when using stdout pipe with Go.

**Review by stffabi — COMMENTED** [review]
That's a great enhancement for the frontend server integration 🚀 

I've left some comments about the stdout handling, which probably might need to be addressed.

File: v2/cmd/wails/internal/commands/dev/dev.go [hunk 2]
```diff
@@ -411,18 +419,22 @@ func loadAndMergeProjectConfig(cwd string, flags *devFlags) (*project.Project, e
...
 	dir := filepath.Join(cwd, "frontend")
 	cmdSlice := strings.Split(devCommand, " ")
 	cmd := exec.CommandContext(ctx, cmdSlice[0], cmdSlice[1:]...)
 	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
 	cmd.Dir = dir
 	setParentGID(cmd)
+	stdo, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v2/cmd/wails/internal/commands/dev/dev.go:432]: Ah, yes 😉

File: v2/cmd/wails/internal/commands/dev/dev.go [hunk 3]
```diff
@@ -411,18 +419,22 @@ func loadAndMergeProjectConfig(cwd string, flags *devFlags) (*project.Project, e
...
+func runFrontendDevWatcherCommand(cwd string, devCommand string, discoverViteServerURL bool) (func(), string, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	dir := filepath.Join(cwd, "frontend")
 	cmdSlice := strings.Split(devCommand, " ")
 	cmd := exec.CommandContext(ctx, cmdSlice[0], cmdSlice[1:]...)
 	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
 	cmd.Dir = dir
 	setParentGID(cmd)
+	stdo, err := cmd.StdoutPipe()
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v2/cmd/wails/internal/commands/dev/dev.go:430]: Ah, yes 😉

File: v2/cmd/wails/internal/commands/dev/dev.go [hunk 4]
```diff
@@ -438,12 +450,32 @@ func runFrontendDevWatcherCommand(cwd string, devCommand string) (func(), error)
 		wg.Done()
 	}()
 
+	var viteServerURL string
+	if discoverViteServerURL {
+		buf := bufio.NewReader(stdo)
+		for {
+			line, _, _ := buf.ReadLine()
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v2/cmd/wails/internal/commands/dev/dev.go:457]: Yeah, originally there was a counter but then that might not reach its limit. Thought about adding some kind of timeout but thought a blocking read would prevent that. We could use a select, a timer and run the current code in a go routine? Any other thoughts?

I saw this as low risk as it is only ever run if you explicitly set it to "auto" and that implies you're using vite. Happy to change it though 👍

**Review by stffabi — APPROVED** [review]
Looks great to me 🚀

**Pull request merged**